### PR TITLE
*: only retry for special message with 1105

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

`1105` means an unknown error in MySQL/TiDB. some of these errors are retryable, but some are not (like `error="Error 1105: statement count 5001 exceeds the transaction limitation`).

### What is changed and how it works?

only retry for retryable errors, now including:
- `Information schema is out of date`
- `Information schema is changed`

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@WangXiangUSTC @amyangfei PTAL

File: syncer/db.go [hunk 1]
```diff
@@ -238,7 +238,7 @@ func (conn *DBConn) executeSQLWithIgnore(tctx *tcontext.Context, ignoreError fun
 				sqlRetriesTotal.WithLabelValues("stmt_exec", conn.cfg.Name).Add(1)
 				return true
 			}
-			if retry.IsRetryableErrorFastFailFilter(err) {
+			if retry.IsRetryableError(err) {
```
Comment (WangXiangUSTC) [inline syncer/db.go]: why change to this function? seems change retry logic for ddl

Comment (csuzhangxc) [reply syncer/db.go]: `IsRetryableErrorFastFailFilter` as a black-list for `1105`, but we have changed to use a white-list for `1105` now.

so, the _fast fail error_ in `IsRetryableErrorFastFailFilter` will not be retry in `IsRetryableError`.

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**